### ISSUE TITLE
Firewall State 2

### ISSUE THREAD

**Issue by heywoodlh** [rgcr/m-cli#160]
Is there a way to enable the firewall with `m` but place it in State 2 (block all incoming connections)?

As far as I am aware `m firewall enable` only puts the firewall in State 1:

```bash
$ m firewall status
Firewall is enabled. (State = 2)
$ m firewall disable && m firewall enable
Firewall is disabled. (State = 0)
Firewall is enabled. (State = 1)
```

Is there a way to set the firewall to State 2?

**Comment by bensleveritt** [comment]
Hey, thanks for filing this issue. I'm not familiar with firewall states, and a cursory investigation reveals that `/usr/libexec/ApplicationFirewall/socketfilterfw` (what this command uses under the hood) doesn't have an obvious way to set state 2 on. I did notice that if I set block all, then turn it off and on from the commandline, it maintained the setting in Settings.

Do you see the same thing?

**Comment by heywoodlh** [comment]
I do not see the same thing. 

This is before I disable and re-enable with `m`:

<img width="666" alt="image" src="https://user-images.githubusercontent.com/18178614/117195230-480c3980-ada2-11eb-84c4-21fcb6164de0.png">

Then I disable and re-enable with `m`, using the following command:

```bash
m firewall disable && m firewall enable
```

After I quit out of System Preferences (it doesn't seem to refresh the settings in realtime in System Preferences if you disable/enable with `m`) and open the firewall tab back up in Security & Privacy, these are my settings:

<img width="666" alt="image" src="https://user-images.githubusercontent.com/18178614/117195498-90c3f280-ada2-11eb-866c-3070287762bd.png">

**Comment by heywoodlh** [comment]
So I looked at the command you mentioned and found that I can set State 2 with the following command:

```bash
sudo /usr/libexec/ApplicationFirewall/socketfilterfw --setblockall on
```

You can see what happens when I disable the firewall with `m`, re-enable the firewall, and then issue that command:
```bash
$ m firewall status
Firewall is enabled. (State = 2)

$ m firewall disable
Firewall is disabled. (State = 0)

$ m firewall enable
Firewall is enabled. (State = 1)

$ m firewall status
Firewall is enabled. (State = 1)

$ sudo /usr/libexec/ApplicationFirewall/socketfilterfw --setblockall on
Firewall is set to block all non-essential incoming connections

$ m firewall status
Firewall is enabled. (State = 2)
```

**Comment by bensleveritt** [comment]
Ah well researched! If you feel up to it, feel free to open a PR. Otherwise I'll have a look at the end of the week.

**Comment by heywoodlh** [comment]
Done!

https://github.com/rgcr/m-cli/pull/161

**Comment by bensleveritt** [comment]
Merged #161 , so I'll close this
